src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -104,7 +124,7 @@ internal static ReportDiagnostic GetDiagnosticReport(DiagnosticSeverity severity
                 location.SourceTree != null &&
                 ((SyntaxTree)location.SourceTree).GetPragmaDirectiveWarningState(id, location.SourceSpan.Start) == ReportDiagnostic.Suppress)
             {
-                return ReportDiagnostic.Suppress;
+                hasPragmaSuppression = true;
```

Comment:
Suppression in source is now not conveyed through the effective severity, but rather through Diagnostic's new property IsSuppressed. We need to retain the effective severity of the suppressed in source diagnostic, as we will be emitting it in the error log and also showing in the error list.